generate_bindings.sh

```diff
@@ -2,23 +2,66 @@
 
 # Always have the latest version of bindgen and rustfmt installed before using this script
 
+set -eu
+
 export DYLD_LIBRARY_PATH=$(rustc +stable --print sysroot)/lib
 
 SDK_VERSION=`xcodebuild -sdk macosx -version SDKVersion`
 SDK_PATH=`xcodebuild -sdk macosx -version Path`
 FRAMEWORK_PATH="$SDK_PATH/System/Library/Frameworks/"
-HEADER_PATH="$FRAMEWORK_PATH/SystemConfiguration.framework/Headers/SCDynamicStore.h"
 
-BINDING_PATH="./system-configuration-sys/src/dynamic_store.rs"
+PREFERENCES_HEADER_PATH="$FRAMEWORK_PATH/SystemConfiguration.framework/Headers/SCPreferences.h"
+DYNAMIC_STORE_HEADER_PATH="$FRAMEWORK_PATH/SystemConfiguration.framework/Headers/SCDynamicStore.h"
+NETWORK_CONFIGURATION_HEADER_PATH="$FRAMEWORK_PATH/SystemConfiguration.framework/Headers/SCNetworkConfiguration.h"
+
+PREFERENCES_BINDING_PATH="./system-configuration-sys/src/preferences.rs"
+DYNAMIC_STORE_BINDING_PATH="./system-configuration-sys/src/dynamic_store.rs"
+NETWORK_CONFIGURATION_BINDING_PATH="./system-configuration-sys/src/network_configuration.rs"
 
 BINDGEN_VERSION=`bindgen --version`
 
 echo "Using macOS SDK at: $SDK_PATH"
 echo "Using $BINDGEN_VERSION"
 echo ""
 
-echo "Generating bindings for $HEADER_PATH"
+echo "Generating bindings for $PREFERENCES_HEADER_PATH"
+bindgen \
+    --no-doc-comments \
+    --whitelist-function "SCPreferences.*" \
+    --blacklist-type "(__)?CF.*" \
+    --blacklist-type "Boolean" \
+    --blacklist-type "dispatch_queue_[ts]" \
+    --blacklist-type "(AuthorizationOpaqueRef|__SCPreferences)" \
+    --raw-line "// Generated using:" \
+    --raw-line "// $BINDGEN_VERSION" \
+    --raw-line "// macOS SDK $SDK_VERSION." \
+    --raw-line "" \
+    --raw-line "use core_foundation_sys::array::CFArrayRef;" \
+    --raw-line "use core_foundation_sys::base::{Boolean, CFIndex, CFAllocatorRef, CFTypeID};" \
+    --raw-line "use core_foundation_sys::data::CFDataRef;" \
+    --raw-line "use core_foundation_sys::string::CFStringRef;" \
+    --raw-line "use core_foundation_sys::propertylist::CFPropertyListRef;" \
+    --raw-line "use core_foundation_sys::runloop::CFRunLoopRef;" \
+    --raw-line "" \
+    --raw-line "use dispatch_queue_t;" \
+    --raw-line "use std::os::raw::c_void;" \
```

Comment:
Use `libc::c_void` is better ?

See https://github.com/mullvad/system-configuration-rs/pull/8/files#diff-037df056d8dd92f60acd1374dc295111R163

if use `libc::c_void` ,  we do not need use `mem::transmute`.

Reply:
You should not have to transmute anyway, a raw pointer can be cast between types anyway. But yes, `libc::c_void` vs `std::os::raw::c_void` can still be discussed.

It's a hassle that `libc` define their own `c_void`, it's identical to the one in `std`. I would very much like if `libc` just did `pub use std::os::raw::c_void` so they became the same type. I'm usually trying to stick with the `std::os::raw` module where possible since that can completely eliminate the `libc` dependency if all one need is the `c_*` types, which are in `std`. But yeah, here we will need `libc` anyway and since `core-foundation` uses `libc` it makes sense to be consistent! I'll change it over to `libc::c_void`!